modules/core/src/main/java/org/apache/fluo/core/impl/TransactionImpl.java

```diff
@@ -972,13 +997,17 @@ private void beginCommitAsync(CommitData cd, AsyncCommitObserver commitCallback,
     final ConditionalMutation pcm =
         prewrite(cd.prow, cd.pcol, cd.pval, cd.prow, cd.pcol, isTriggerRow(cd.prow));
 
-    ListenableFuture<Iterator<Result>> future = cd.acw.apply(Collections.singletonList(pcm));
-    Futures.addCallback(future, new CommitCallback<Iterator<Result>>(cd) {
-      @Override
-      protected void onSuccess(CommitData cd, Iterator<Result> result) throws Exception {
-        postLockPrimary(cd, pcm, Iterators.getOnlyElement(result));
-      }
-    }, env.getSharedResources().getAsyncCommitExecutor());
+    /*
+     * ListenableFuture<Iterator<Result>> future = cd.acw.apply(Collections.singletonList(pcm));
+     * Futures.addCallback(future, new CommitCallback<Iterator<Result>>(cd) {
+     * 
+     * @Override protected void onSuccess(CommitData cd, Iterator<Result> result) throws Exception {
+     * postLockPrimary(cd, pcm, Iterators.getOnlyElement(result)); } },
+     * env.getSharedResources().getAsyncCommitExecutor());
+     */
+    CompletableFuture<Iterator<Result>> cfuture = cd.acw.apply(Collections.singletonList(pcm));
+    addCallback(cfuture, cd, (cd2, result) -> postLockPrimary(cd2, pcm,
+        Iterators.getOnlyElement((Iterator<Result>) result)));
```

Comment:
Hopefully this type cast `(Iterator<Result>)` can be dropped after adding the type param to addCallback